monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -0,0 +1,370 @@
+        """
+        self.fill_template_config(self.data_stats_files, **kwargs)
+        write_path = os.path.join(output_path, algo_name)
+        self.cfg["bundle_root"] = write_path
+        os.makedirs(write_path, exist_ok=True)
+        # handling scripts files
+        output_scripts_path = os.path.join(write_path, "scripts")
+        if os.path.exists(output_scripts_path):
+            shutil.rmtree(output_scripts_path)
+        if self.scripts_path is not None and os.path.exists(self.scripts_path):
+            shutil.copytree(self.scripts_path, output_scripts_path)
+        # handling config files
+        output_config_path = os.path.join(write_path, "configs")
+        if os.path.exists(output_config_path):
+            shutil.rmtree(output_config_path)
+        os.makedirs(output_config_path, exist_ok=True)
+        output_config_file = os.path.join(output_config_path, "algo_config.yaml")
+        ConfigParser.export_config_file(self.cfg.config, output_config_file, fmt="yaml", default_flow_style=None)
+        with open(output_config_file, "r+") as f:
+            lines = f.readlines()
+            f.seek(0)
+            f.write(f"# Generated automatically by `{__name__}`\n")
+            f.write("# For more information please visit: https://docs.monai.io/\n\n")
+            for item in ensure_tuple(self.template_configs):
+                f.write(f"# source file: {item}\n")
+            f.write("\n\n")
+            f.writelines(lines)
+        logger.info(write_path)
+        self.output_path = write_path
+
+    def train(self, train_params=None):
+        """
+        Load the run function in the training script of each model. Training parameter is predefined by the
+        algo_config.yaml file, which is pre-filled by the fill_template_config function in the same instance.
+
+        Args:
+            train_params:  to specify the devices using a list of integers: ``{"CUDA_VISIBLE_DEVICES": [1,2,3]}``.
+        """
+        if train_params is not None:
+            params = deepcopy(train_params)
+
+        train_py = os.path.join(self.output_path, "scripts", "train.py")
+        config_yaml = os.path.join(self.output_path, "configs", "algo_config.yaml")
+        base_cmd = f"{train_py} run --config_file={config_yaml} "
+
+        if "CUDA_VISIBLE_DEVICES" in params:
+            devices = params.pop("CUDA_VISIBLE_DEVICES")
+            n_devices, devices_str = len(devices), ",".join([str(x) for x in devices])
+        else:
+            n_devices, devices_str = torch.cuda.device_count(), ""
+        if n_devices > 1:
+            cmd = f"torchrun --nnodes={1:d} --nproc_per_node={n_devices:d} "
+        else:
+            cmd = "python "  # TODO: which system python?
+        cmd += base_cmd
+        if params and isinstance(params, Mapping):
+            for k, v in params.items():
+                cmd += f" --{k}={v}"
```

Comment:
not sure how good it is for passing args with special characters... may need some tests for file paths.

Reply:
I did some simple tests. It seems overall consistent with Python behavior ("aaa/bbb", "aaa\\bbb" okay but not "a\b"). Such test seems standalone from the ensemble module and we can construct something else to test the subprocess call.